File: src/main/java/ninetynineproblems/lists/P01.java
package ninetynineproblems.lists;

import java.util.List;
import java.util.NoSuchElementException;

public class P01 {

  public static String last(List<String> list) {
    if (list == null || list.isEmpty()) {
      throw new NoSuchElementException();
    }
    if (list.size() == 1) {
      return list.get(0);
    }
    return last(list.subList(1, list.size()));
  }
}

File: src/main/kotlin/ninetynineproblems/lists/P01.kt
package ninetynineproblems.lists

import java.util.*

fun last(list: List<String>?): String {
  if (list == null || list.isEmpty()) {
    throw NoSuchElementException()
  }
  if (list.size == 1) {
    return list[0]
  }
  return last(list.subList(1, list.size))
}
